lib/CustomMySQL.js

```diff
@@ -159,6 +159,13 @@ var CustomMySQL = function () {
                 this._cache[0] = query.text;
                 this._cache.splice(1, 0, query.values);
             }
+            // Check if knex instance
+            else if (this._cache[0] && this._cache[0].toSQL) {
```

Comment:
MINOR: This is fairly generic, we don't have anything more specific to knex?

Reply:
knex doesn't have specific api to use for determining if it's instance of knex unlike the squel's `isSquelBuilder`

Reply:
Can we check for class/instances etc?

Reply:
Not really, it's just an object: https://github.com/knex/knex/blob/master/lib/knex-builder/Knex.js

```
'use strict';

const knex = require('knex')({
    client: 'mysql',
});

const query = knex
	.select('name')
	.from('users')
	.where('name', 'name');


console.log(typeof query); // object
console.log(query instanceof knex); // false
```

Reply:
`knex` is instanceOf `Builder` declare inside `knex/lib/query/querybuilder`
but the require to it needs to be wrapped in try catch to avoid crashing if user don't have knex installed as dependency.


```js
'use strict';
let Builder

// try catch to not crash if knex is not installed
try {
  Builder = require('knex/lib/query/querybuilder')
} catch {
}

const knex = require('knex')({
    client: 'mysql',
});


const query = knex
	.select('name')
	.from('users')
	.where('name', 'name');

console.log(Builder && query instanceof Builder); // true
```